### PR TITLE
feat: Add mutation for `fan_mode_setting`

### PR DESCRIPTION
**Pull request by xplato**
~~Blocked by #559~~

### CODE REVIEW

File: src/lib/seam/thermostats/use-update-fan-mode.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export function useUpdateFanMode(): UseMutationResult<
     },
     onSuccess: (_data, variables) => {
       queryClient.setQueryData<ThermostatDevice | null>(
-        ['thermostats', 'get', { device_id: variables.device_id }],
+        ['devices', 'get', { device_id: variables.device_id }],
```
Comment (razor-x) [inline src/lib/seam/thermostats/use-update-fan-mode.ts:39]: Nice catch. Any other ones with this issue?

File: src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx [hunk 2]
```diff
@@ -106,6 +111,17 @@ export function ThermostatDeviceDetails({
                   temperatureUnit='fahrenheit'
                 />
               </DetailRow>
+              <DetailRow label={t.fanMode}>
+                <FanModeMenu
+                  mode={device.properties.fan_mode_setting ?? 'auto'}
```
Comment (razor-x) [inline src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx:116]: Why is the default auto? Is this just an issue with the types or can the fan mode really be undefined on the API?

**Review by razor-x — APPROVED** [review]

File: src/lib/seam/thermostats/use-update-fan-mode.ts (hunk 1 above)
Comment (xplato) [reply src/lib/seam/thermostats/use-update-fan-mode.ts:39]: Yes, I decided to do this in a separate PR for the other hook: #562

File: src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx (hunk 2 above)
Comment (xplato) [reply src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx]: Ahh thank you—this is actually not required (forgot to remove this after resolved underlying type errors)

**Pull request merged**